index.rst

```diff
@@ -1,47 +1,159 @@
+
+    raise ImportError("This package is being renamed to pipe_best! Do not use!")
+
+Commit those changes and push your branch.
+
+This will help you identify packages that depend on ``meas_worst``, as they will break as soon as they attempt to import any portion of it.
+
+Branch dependent packages
+-------------------------
+
+Create ``tickets/DM-NNNN`` branches for those packages you know are dependent on ``meas_worst``. Any that you weren't aware of will be discovered when you run Jenkins in a moment.
+
+Update dependent packages
+-------------------------
+
+In each of the dependent packages:
+
+* Update the ups dependencies in the table file: ``setupRequired(meas_worst)`` -> ``setupRequired(pipe_best)``.
+* Change any python import statements and namespaces, and any C++ include files and namespaces (top-level or explicit).
+* Commit and push your branch.
+
+Test in Jenkins
+---------------
+
+You can test the individual dependent packages one at a time, but you also need to test everything in Jenkins to ensure you haven't missed any dependent packages. Because you've added ``pipe_best`` to :file:`repos.yaml` on master and done all of the above work on one ticket branch, you can submit a Jenkins job for your branch and it will test all of your changes, plus it will make it clear if you've missed anything because of the ``ImportError`` statements in ``meas_worst``.
+
+Update confluence package history page
+--------------------------------------
+
+Add another entry to the `confluence package history`_ table, noting the date you expect the code review and merge for this package rename to be complete.
+
+Get reviewed
+------------
+
+Once Jenkins passes, including all the demos, and you've pushed your changes to all of the dependent package branches, have your ticket reviewed. This step should not be too difficult for the reviewer, even though many packages have changed, as the individual changes should be small. You can refer the reviewer to this document for them to refer to during the review (e.g. to prevent file renames and content changes being part of the same commit).
+
+Post on Community and Slack
+---------------------------
+
+To ensure that other developers are aware of the pending change, post to the appropriate rooms on Slack (e.g. `dm`) and write up a short Community_ post describing the change and any caveats that other developers should be aware of.
+
+.. _Community: https://community.lsst.org
+
+Merging in work that had started on the old package
+===================================================
+
+Once your rename has been merged to master, other developers may have open branches on ``meas_worst`` that they will want to move to ``pipe_best``. Because you did the various steps above as individual commits, they should be able to rebase cleanly. Those developers can follow the following steps:
+
+1. Push your ``meas_worst`` changes to github on your branch.
```

Comment:
This section got significantly rewritten yesterday afternoon. Please reload the page to see the latest version.